app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/VpnInternalSettingsActivity.kt

```diff
@@ -82,6 +86,14 @@ class VpnInternalSettingsActivity : DuckDuckGoActivity() {
         }.also { sendBroadcast(it) }
     }
 
+    private val cpuMonitoringToggleListener = CompoundButton.OnCheckedChangeListener { _, toggleState ->
+        if (toggleState) {
+            cpuMonitor.startMonitoring()
+        } else {
+            cpuMonitor.stopMonitoring()
+        }
+    }
```

Comment:
This is OK but I think it may have one issue, ie the monitor will continue getting CPU stats when the user disables the VPN. And probably we don't want that?

I would do this through a local feature flag, you can take a look at `AppTpSetting#AlwaysSetDNS` as it'd be very similar.

Reply:
Great catch - fixed!